### PR TITLE
Refactor: Add A400 Laser 10W camera capture workflow & manual connect WIFI

### PR DESCRIPTION
**Pull request by xiekun1992**
[LU-2292](https://snapmaker2.atlassian.net/browse/LU-2292)

### CODE REVIEW

**Comment by wen-qing-255** [comment]
包括以下参数修改 https://snapmaker2.atlassian.net/browse/LU-2278

File: src/server/services/socket/SACP-TCP.ts [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+    }
+
+    public refreshDevices = () => {
+        wifiServerManager.refreshDevices();
+    }
+
+    public connectionOpen = (socket: SocketServer, options: EventOptions) => {
+        this.socket = socket;
+        this.sacpClient.setHandler(0x01, 0x03, (data) => {
+            const state: any = {
```
Comment (MaShizhen) [inline src/server/services/socket/SACP-TCP.ts:65]: should be banned

File: src/server/services/socket/SACP-TCP.ts [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+            };
+            this.socket && this.socket.emit('connection:open', result);
+        });
+    }
+
+    public connectionClose = (socket: SocketServer, options: EventOptions) => {
+        this.client.destroy();
+        if (this.client.destroyed) {
+            log.info('TCP manually closed');
+            const result: any = {
```
Comment (MaShizhen) [inline src/server/services/socket/SACP-TCP.ts:145]: should be banned

**Review by MaShizhen — CHANGES_REQUESTED** [review]

File: src/server/lib/image-getPhoto.js [hunk 3]
```diff
@@ -3,109 +3,203 @@ import Jimp from 'jimp';
...
+    } else {
+        let api = `http://${address}:8080/api/request_capture_photo`;
+        api += `?index=${index}&x=${x}&y=${y}&z=${z}&feedRate=${feedRate}`;
+        if (photoQuality >= 0 && photoQuality <= 255) {
+            api += `&photoQuality=${photoQuality}`;
+        }
+        return new Promise((resolve) => {
+            request.get(api).end((err, res) => {
+                log.debug(`request_capture_photo: ${res}`);
+                resolve({
```
[the remark this answers is not in the document]
Comment (xiekun1992) [reply src/server/lib/image-getPhoto.js:44]: only debug info

File: src/app/ui/widgets/WifiTransport/WifiTransport.jsx [hunk 4]
```diff
@@ -472,9 +473,8 @@ function WifiTransport({ widgetActions, controlActions }) {
     }, [selectFileName]);
 
     useEffect(() => {
-        const newCurrent = connectionType === 'wifi' ? workflowStatus : workflowState;
-        setCurrentWorkflowStatus(newCurrent);
-    }, [workflowState, workflowStatus, connectionType]);
+        setCurrentWorkflowStatus(workflowStatus);
+    }, [workflowStatus]);
```
Comment (jane-rose) [inline src/app/ui/widgets/WifiTransport/WifiTransport.jsx:477]: checkout this for now

File: src/server/services/socket/ConnectionManager.js [hunk 5]
```diff
@@ -64,6 +85,44 @@ class ConnectionManager {
         this.socket && this.socket.connectionClose(socket, options);
     };
 
+    inspectProtocol = async (address) => {
+        const PORT_SCREEN_HTTP = 8080, PORT_SCREEN_SACP = 8888;
```
Comment (jane-rose) [inline src/server/services/socket/ConnectionManager.js:89]: make this a constant

File: src/server/services/task-manager/workers/generateGcode.js [hunk 6]
```diff
@@ -78,10 +78,11 @@ const checkoutBoundingBoxIsNull = (boundingBox) => {
 };
 
 // eslint-disable-next-line consistent-return
-const generateGcode = (toolPaths) => {
+const generateGcode = ({ toolPaths, size, toolHead, origin, series }) => {
     if (!toolPaths && !_.isArray(toolPaths) && toolPaths.length === 0) {
         return sendMessage({ status: 'fail', value: 'modelInfo is empty.' });
     }
+    console.log({ size });
```
Comment (jane-rose) [inline src/server/services/task-manager/workers/generateGcode.js:85]: log

**Review by jane-rose — APPROVED** [review]

**Review by wen-qing-255 — APPROVED** [review]

**Pull request merged**